### PR TITLE
Interactive-branch

### PR DESCRIPTION
**Pull request by kayjeee**
Added a new JavaScript file and import it as a module:
it will contain methods related to the status updates (completed: true / false).
Added event listener to the checkbox (change).
Updated items object's value for completed key upon user actions.
Implemented a function for the "Clear all completed" button (use filter() method).
Stored the updated array of items in local storage, so the user gets the correct list of values after the page reloads (which means that any changes made to the list should be preserved).

### CODE REVIEW

File: src/index.js [hunk 1]
```diff
@@ -12,4 +12,4 @@ addTodo();
 deletetodo();
 edittodo();
 interacttodo();
-clearcompleted();
\ No newline at end of file
+clearcompleted();
```
Comment (AnselemOdims) [inline src/index.js]: - [ ] While some of the functionalities work, after marking an item as completed, the checkbox is not checked which does not indicate to the user the item has been completed. Please kindly fix this.

image

**Review by AnselemOdims — CHANGES_REQUESTED** [review]
### Hi @kayjeee , 
You have been doing a tremendous job so far :ok_hand:
There are some issues that you still need to work on to go to the next project but you are almost there!

## To Highlight 🎉
PR has a title and summary. :heavy_check_mark:
No linter errors :heavy_check_mark:

## Required Changes ♻️
- [ ] Please check inline comments under the review

## Optional suggestions
_Comments with the [OPTIONAL] prefix are not a must to implement however, It's strongly recommended to work on them as they can make your code better._

## Happy coding!👏👏👏

#### Please leave any questions or comments in the PR if you have any doubts or unclarity! You can tag me using @AnselemOdims for a quicker response

**_Note: Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s).

**Comment by AnselemOdims** [comment]
PS: I reviewed this project because at the time GitHub Actions were down, but linters were passing locally

File: src/index.js (hunk 1 above)
Comment (TadesseDev) [inline src/index.js]: - [x] Dear @kayjeee, Just as mentioned by the previous reviewer, Your task is not getting checked once the user mark it as completed. This is unexpected behavior according to the requirements. Please make sure the `checkbox` is ticked/✅once the user marks it as completed. 

https://github.com/kayjeee/webpack-todolist/assets/69077061/a7fbe66b-1abc-4de8-b3b0-62a35316c955

**Review by TadesseDev — CHANGES_REQUESTED** [review]
### Hi @kayjeee 

### I'm Tadesse Alemayehu your reviewer for this PR. :hand: 

Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.

199733072-bdacca19-8df9-43a7-be93-3aff2f1c15c8

 ## Highlights
 - [x] Good job making most of the changes requested by the previous reviewer. 🙏


 ## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/index.js (hunk 1 above)
Comment (kayjeee) [reply src/index.js]: Dear @TadesseDev 

I just wanted to take a moment to express my gratitude for your recent review feedback on my work. Your valuable comments and suggestions have been extremely helpful in improving my work and making it better.
I can see exactly what you mean ,I believe I have it fixed now

Comment (TadesseDev) [reply src/index.js]: Great to know you found value in my feedback. Good job @kayjeee.

**Review by AbelG101 — APPROVED** [review]
# Hi @kayjeee 👋, 

**Your project is complete!  There is nothing else to say other than... it's time to merge it. 
Congratulations! 🎉 🥳🥇**

## Highlights
- Good communication with reviewers ✔
-  Unique UI ✔
- All linter checks are passing ✔
- Descriptive PR title with a great summary ✔

## Optional suggestions
- N/A

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**